newIDE/app/src/Export/LocalExporters/LocalPreviewLauncher/index.js

```diff
@@ -149,13 +150,22 @@ export default class LocalPreviewLauncher extends React.Component<
         new gd.AbstractFileSystemJS(),
         localFileSystem
       );
-      const outputDir = path.join(fileSystem.getTempDir(), 'preview');
-      const exporter = new gd.Exporter(fileSystem, gdjsRoot);
+      try {
+         const outputDir = path.join(fileSystem.getTempDir(), `preview-${os.userInfo().username}`);
+         const exporter = new gd.Exporter(fileSystem, gdjsRoot);
+         return {
+           outputDir,
+           exporter,
+         };
```

Comment:
Avoid repeating too much code, just use the try catch around the code that is requiring it. Put it in a function if it's better for the readability :)

Reply:
I did it at first but this caused issues so I was forced to do like so.